motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -944,9 +977,41 @@ BOOL Ros_MotionServer_StartTrajMode(Controller* controller)
 	if(Ros_Controller_IsServoOn(controller) == FALSE)
 	{
 		MP_SERVO_POWER_SEND_DATA sServoData;
-		memset(&rData, 0x00, sizeof(rData));
 		memset(&sServoData, 0x00, sizeof(sServoData));
+
+		if (Ros_Controller_IsEcoMode(controller) == TRUE)
+		{
+			//toggle servos to disable energy-savings mode
+			sServoData.sServoPower = 0;  // OFF
+			memset(&rData, 0x00, sizeof(rData));
+			ret = mpSetServoPower(&sServoData, &rData);
+			if ((ret == 0) && (rData.err_no == 0))
+			{
+				// wait for the Servo/Eco OFF confirmation
+				int checkCount;
+				for (checkCount = 0; checkCount<MOTION_START_TIMEOUT; checkCount += MOTION_START_CHECK_PERIOD)
+				{
+					// Update status
+					Ros_Controller_StatusUpdate(controller);
+
+					if (Ros_Controller_IsEcoMode(controller) == FALSE)
+						break;
+
+					mpTaskDelay(MOTION_START_CHECK_PERIOD);
+				}
+			}
+			else
+			{
+				char errMsg[ERROR_MSG_MAX_SIZE];
+				memset(errMsg, 0x00, ERROR_MSG_MAX_SIZE);
+				Ros_Controller_ErrNo_ToString(rData.err_no, errMsg, ERROR_MSG_MAX_SIZE);
+				printf("Can't disable energy-savings mode because: %s\r\n", errMsg);
+				goto updateStatus;
+			}
+		}

```

Comment:
:)

I actually meant the entire piece of code starting with 

```c++
if (servoOnOff == ON && Ros_Controller_IsEcoMode(controller) == TRUE)
```

not just the error printing bit :)

but that is also nice to put into a separate function.